### PR TITLE
docs: add tslib update migration docs

### PR DESCRIPTION
**Pull request by clydin**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?

This adds documentation for the tooling migration `update-libraries-tslib`.
Migration code PR: https://github.com/angular/angular-cli/pull/17759

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by alan-agius4 — APPROVED** [review]
LGTM, very excellent explanation.

Eventually we should add a link to here from the migration.

**Review by aikithoughts — COMMENTED** [review]
This should also be added to the navigation.json file so users can find it, correct?

**Comment by clydin** [comment]
@aikidave There will be direct links to the document from within the migration descriptions shown to the user when the update process is executed.  I think these can also be added in the "Updating to V10" section once that lands.

**Review by IgorMinar — APPROVED** [review]
I rebased this PR, rewrote several sections, and force-pushed the changes.

@clydin can you please take a look if you are ok with my changes.
@aikidave can you do one more read through this please as well? thanks!

File: aio/content/guide/migration-update-libraries-tslib.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+## Why is this migration necessary?
+
+The [`tslib`](https://github.com/Microsoft/tslib) is a runtime library for Typescript.
+The version of this library is bound to the version of the TypeScript compiler used to compile a library.
+Peer dependencies do not accurately represent this relationship between the runtime and the compiler.
+If `tslib` remained declared as a library peer dependency, it would be possible for some Angular workspaces to get into a state where the workspace could not satisfy `tslib` peer dependency requirements for multiple libraries, resulting in build-time or run-time errors.
+
+As of TypeScript 3.9 (used by Angular v10), `tslib` version of 2.x is required to build new applications.
+However older libraries built with previous version of TypeScript and already published to npm might need `tslib` 1.x.
```
Comment (aikidave) [inline aio/content/guide/migration-update-libraries-tslib.md:45]:
```suggestion
However, older libraries built with previous version of TypeScript and already published to npm might need `tslib` 1.x.
```

File: aio/content/guide/migration-update-libraries-tslib.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+This migration makes it possible for code depending on these incompatible versions of `tslib` to remain interoperable.
```
Comment (aikidave) [inline aio/content/guide/migration-update-libraries-tslib.md:46]: Not required, but it seems like this sentence is what really explains why this migration is required. You could move it to the beginning of this section, and reword it slightly:

```suggestion
This migration makes it possible for code depending on incompatible versions of the `tslib` runtime library to remain interoperable.
```

**Review by aikithoughts — APPROVED** [review]
Reviewed-for: global-docs-approvers

**Pull request closed without merging**